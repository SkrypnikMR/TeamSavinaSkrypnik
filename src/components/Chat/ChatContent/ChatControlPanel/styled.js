import styled from 'styled-components';
import { bgColorDefaultFon, textColorBlack } from '/src/components/UI/baseLayout';


export const StChatControlPanel = styled.div`
    display: flex;
    width: 100%;
    height: 100px;
    padding: 0;
    justify-content: space-between;
    align-items: center;
    color: ${({ color = textColorBlack }) => color};
    font-size: 24px;
    font-family: 'Play', sans-serif;
    border-right: 1px solid black;
    background-color: ${({ color = bgColorDefaultFon }) => color};
`;
