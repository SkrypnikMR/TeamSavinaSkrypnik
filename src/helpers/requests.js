import { routes } from '../constants/routes';

const { baseUrl } = routes;

export const getRequest = async (url, optionalHeaders = {}) => {
  const headers = Object.assign({}, optionalHeaders);
  const options = { method: 'GET', headers };
  const answer = await fetch(`${baseUrl}${url}`, options);
  if (!answer.ok) throw new Error(await answer.json());
  const result = await answer.json();
  return result;
};

export const postRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'POST',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};

export const putRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'PUT',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(`${baseUrl}${url}`, options);
  const result = await answer.json();
  return result;
};
export const createAuthHeader = (support) => { return { Authorization: support.getCookie('token') }; };
