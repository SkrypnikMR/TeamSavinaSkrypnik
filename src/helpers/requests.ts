import { routes } from '../constants/routes';
import { TUrl, TBody } from './types/requestTypes';

const { baseUrl } = routes;

export const getRequest = async (url: TUrl) => {
  const options = { method: 'GET' };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};

export const postRequest = async (url : TUrl, body : TBody) : Promise<Response> => {
      const options = {
              method: 'POST',
              body: JSON.stringify(body),
              headers: { 'Content-Type': 'application/json' },
      };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};
