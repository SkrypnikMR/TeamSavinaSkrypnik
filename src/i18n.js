import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import en from './locales/en.json';
import ar from './locales/ar.json';
import ru from './locales/ru.json';

const language = localStorage.getItem('lang') || 'en';

i18n.use(initReactI18next).init({
    resources: {
        en: { translations: en },
        ar: { translations: ar },
        ru: { translations: ru },
    },
    fallbackLng: ['en', 'ar', 'ru'],
    lng: language,
    debug: false,
    ns: ['translations'],
    defaultNS: 'translations',
    keySeparator: false,
    interpolation: {
        escapeValue: false,
        formatSeparator: ',',
    },
    react: {
        wait: true,
    },
});

export default i18n;
