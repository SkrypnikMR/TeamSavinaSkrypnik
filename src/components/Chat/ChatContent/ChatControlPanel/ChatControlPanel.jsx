import React from 'react';
import PropTypes from 'prop-types';
import { INPUT__MESSAGE } from '/src/constants/componentsСonsts.js';
import { StChatControlPanel } from './styled';
import Input from '/src/components/UI/Input';
import Button from '/src/components/UI/Button';

const ChatControlPanel = (
  { onChangeInput, messageInputValue, sendNewMessage }) => {
  const handleOnclick = () => {
    sendNewMessage();
    onChangeInput({ name: 'newMessage', value: '' });
  };
  const onKeyUpEnter = () => {
    sendNewMessage();
    onChangeInput({ name: 'newMessage', value: '' });
  };
  return (
    <StChatControlPanel>
      <Input
        id={INPUT__MESSAGE.id}
        height="60px"
        key={INPUT__MESSAGE.id}
        name="newMessage"
        inputHeight="60px"
        borderRadius="0px"
        value={messageInputValue}
        onChange={onChangeInput}
        placeholder='placeholder_control_input'
        margin="0 auto"
        bgColor="transparent"
        color="white"
        onKeyUpEnter={onKeyUpEnter}
        fontSizeInp="20px"
        borderColor="transparent"
        bgFocusColor="transparent"
        padding="20px"
      />
      <Button
        height="50px"
        width="50px"
        content="✈"
        fontSize="40px"
        borderRadius="0px"
        margin="0px"
        padding="0 60px 0 0"
        id="sendMessage"
        onClick={handleOnclick}
        bgColor="transparent"
      />
    </StChatControlPanel>
  );
};

ChatControlPanel.propTypes = {
  onChangeInput: PropTypes.func,
  sendNewMessage: PropTypes.func,
  messageInputValue: PropTypes.string,
};

export default ChatControlPanel;
