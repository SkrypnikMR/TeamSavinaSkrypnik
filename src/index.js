import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import rootReducer from './Store';
import App from './components/App';
import rootSaga from './Store/saga';
import './i18n';

const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(saga),
    devTools,
  ),
);
saga.run(rootSaga);

render(
  <Provider store={store}>
    <App />
  </Provider >,
  document.getElementById('root'),
);
