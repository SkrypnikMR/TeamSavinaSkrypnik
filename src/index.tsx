import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import rootReducer from './store';
import App from './components/App';
import rootSaga from './store/saga';
import './i18n';

const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(saga),
    devTools,
  ),
);
saga.run(rootSaga);

render(
  <Provider store={store}>
    <App />
  </Provider >,
  document.getElementById('root'),
);
