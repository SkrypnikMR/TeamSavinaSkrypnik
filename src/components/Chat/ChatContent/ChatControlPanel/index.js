import { connect } from 'react-redux';
import ChatControlPanel from './ChatControlPanel.jsx';
import { setValue, sendNewMessage } from '/src/Store/chat/actions';
import { newMessage } from '/src/Store/chat/selectors';

const mapStateToProps = state => ({
    messageInputValue: newMessage(state),
});

const mapDispatchToProps = dispatch => ({
    onChangeInput: payload => dispatch(setValue(payload)),
    sendNewMessage: () => dispatch(sendNewMessage()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatControlPanel);
