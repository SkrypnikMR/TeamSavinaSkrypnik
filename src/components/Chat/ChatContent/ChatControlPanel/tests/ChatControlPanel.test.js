import React from 'react';
import { shallow } from 'enzyme';
import ChatControlPanel from '../ChatControlPanel';
import { shallowSmart, mountSmart } from '/src/helpers/testHelper';
import '/src/i18n';

describe('ChatControlPanel', () => {
  let props;
  const onChangeInput = jest.fn();
  const messageInputValue = '';
  const sendNewMessage = jest.fn();
  beforeEach(() => {
    props = {
      onChangeInput,
      messageInputValue,
      sendNewMessage,
    };
  });
  it('Should match snapshot', () => {
    const component = shallowSmart(<ChatControlPanel />);
    expect(component.html).toMatchSnapshot();
  });
  it('should render inputs', () => {
    const component = mountSmart(<ChatControlPanel {...props} />);
    expect(component.find('Input')).toHaveLength(1);
  });
  it('should render button', () => {
    const component = mountSmart(<ChatControlPanel {...props} />);
    expect(component.find('Button')).toHaveLength(1);
  });
  it('should click on the button', () => {
    const component = mountSmart(<ChatControlPanel {...props} />);
    component.find('Button').simulate('click');
    expect(sendNewMessage).toHaveBeenCalled();
  });
  it('Should render StChatControlPanel', () => {
    const component = shallow(<ChatControlPanel />);
    expect(component.find('styled__StChatControlPanel')).toHaveLength(1);
  });
});
