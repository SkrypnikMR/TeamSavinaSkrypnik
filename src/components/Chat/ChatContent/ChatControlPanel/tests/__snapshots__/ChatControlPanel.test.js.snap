// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatControlPanel Should match snapshot 1`] = `[Function]`;
